rust/src/io/writer.rs

```diff
@@ -129,18 +129,27 @@ impl<'a> FileWriter<'a> {
     /// All RecordBatch will be treated as one RecordBatch on disk
     ///
     /// Returns [Err] if the schema does not match with the batch.
-    pub async fn write(&mut self, batch: &[&RecordBatch]) -> Result<()> {
-        for field in self.schema.fields.iter() {
-            let arrs: Result<Vec<_>> = batch
+    pub async fn write(&mut self, batches: &[&RecordBatch]) -> Result<()> {
+        // Copy a list of fields to avoid borrow checker error.
+        let fields = self
```

Comment:
I don't understand why tho the code doesn't seem to modify the fields?

Reply:
Oh,  so `self.schema` has a immuntable reference to `self`, but later it needs mutable reference `self` to call `self.write_array`

Reply:
Leave a comment here, in case some Rust guru knows how to do it better later.